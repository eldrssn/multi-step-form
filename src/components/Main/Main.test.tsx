import { render } from '@testing-library/react';
import { Main } from './';

test('initial step state is 1', () => {
  const { getByText, container } = render(<Main />);
  const initialStepText = getByText('Step 1');
  expect(initialStepText).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
