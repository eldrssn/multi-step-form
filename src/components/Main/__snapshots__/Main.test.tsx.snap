// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial step state is 1 1`] = `
<div>
  <section
    class="section"
  >
    <aside
      class="container"
    >
      <div
        class="box box_active"
      >
        <div
          class="number"
        >
          1
        </div>
        <div
          class="info"
        >
          <p
            class="info_number"
          >
            Step 
            1
          </p>
          <p
            class="info_title"
          >
            Your info
          </p>
        </div>
      </div>
      <div
        class="box"
      >
        <div
          class="number"
        >
          2
        </div>
        <div
          class="info"
        >
          <p
            class="info_number"
          >
            Step 
            2
          </p>
          <p
            class="info_title"
          >
            Select plan
          </p>
        </div>
      </div>
      <div
        class="box"
      >
        <div
          class="number"
        >
          3
        </div>
        <div
          class="info"
        >
          <p
            class="info_number"
          >
            Step 
            3
          </p>
          <p
            class="info_title"
          >
            Add-ons
          </p>
        </div>
      </div>
      <div
        class="box"
      >
        <div
          class="number"
        >
          4
        </div>
        <div
          class="info"
        >
          <p
            class="info_number"
          >
            Step 
            4
          </p>
          <p
            class="info_title"
          >
            Summary
          </p>
        </div>
      </div>
    </aside>
    <div
      class="container"
    >
      <div
        class="form_box"
      >
        <h1
          class="title"
        >
          Personal info
        </h1>
        <p
          class="description"
        >
          Please provide your name, email address, and phone number.
        </p>
        <form
          class="form"
        >
          <div
            class="input"
          >
            <div
              class="label_box"
            >
              <label
                for="name"
              >
                Name
              </label>
            </div>
            <input
              id="name"
              label="Name"
              name="name"
              placeholder="e.g. Stephen King"
              type="text"
              value=""
            />
          </div>
          <div
            class="input"
          >
            <div
              class="label_box"
            >
              <label
                for="email"
              >
                Email Address
              </label>
            </div>
            <input
              id="email"
              label="Email Address"
              name="email"
              placeholder="e.g. stephenking@lorem.com"
              type="text"
              value=""
            />
          </div>
          <div
            class="input"
          >
            <div
              class="label_box"
            >
              <label
                for="phone"
              >
                Phone Number
              </label>
            </div>
            <input
              id="phone"
              label="Phone Number"
              name="phone"
              placeholder="e.g. +1 234 567 890"
              type="number"
              value=""
            />
          </div>
          <div
            class="buttons"
          >
            <button
              class="button button_next"
              type="submit"
            >
              Next Step
            </button>
          </div>
        </form>
      </div>
    </div>
  </section>
</div>
`;
